### PR TITLE
refactor: auto-determine stable version

### LINKED ISSUE #1734: Update handling of next branch detection and database namespacing
## Detailed Description

Currently we attempt to detect if the user is running on master or next by looking at the current app version. Then we use this to namespace the database in order to have separate databases for master vs next.

Ideally this should be based on something other than the version string so that we don't have an issue when the version string is incremented.

### PR DESCRIPTION
**Pull request by mrfelton**
## Description:

Use a semver range check to determine weather the current app version is considered to be a stable version.

## Motivation and Context:

Fix #1734

## How Has This Been Tested?

Manually + test suite

## Types of changes:

Refactor

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have reviewed and updated the documentation accordingly.
- [x] I have read the _CONTRIBUTING_ document.
- [x] I have added tests to cover my changes where needed.
- [x] All new and existing tests passed.
- [x] My commits have been squashed into a concise set of changes.

### CODE REVIEW

File: test/unit/utils/isStableVersion.spec.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      ['0.2.0-alpha', STABLE_VERSION, false],
+      ['0.2.0', STABLE_VERSION, false],
+      ['0.4.0-alpha', STABLE_VERSION, false],
+      ['0.4.0-beta', STABLE_VERSION, false]
+    ]
+
+    negativeTests.map(test => {
+      const version = test[0]
+      const stable = test[1]
+      const expectation = test[2]
```
Comment (korhaliv) [inline test/unit/utils/isStableVersion.spec.js]: Consider using es6 destructuring to keep this more compact

File: test/unit/utils/isStableVersion.spec.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      ['0.3.0-beta', STABLE_VERSION, true],
+      ['0.3.1-beta', STABLE_VERSION, true],
+      ['0.3.2-alpha', STABLE_VERSION, true],
+      ['0.3.0', STABLE_VERSION, true]
+    ]
+
+    positiveTests.map(test => {
+      const version = test[0]
+      const stable = test[1]
+      const expectation = test[2]
```
Comment (korhaliv) [inline test/unit/utils/isStableVersion.spec.js]: Consider using es6 destructuring to keep this more compact

File: test/unit/utils/isStableVersion.spec.js (hunk 1 above)
Comment (mrfelton) [reply test/unit/utils/isStableVersion.spec.js]: Updated

File: test/unit/utils/isStableVersion.spec.js (hunk 2 above)
Comment (mrfelton) [reply test/unit/utils/isStableVersion.spec.js]: Updated

**Pull request merged**